cypress/integration/chi-vue/tooltip.spec.js

```diff
@@ -0,0 +1,150 @@
+const TOOLTIP_CLASSES = {
+  TOOLTIP_ELEMENT: 'chi-tooltip'
+};
+const TOOLTIP_EVENTS = {
+  SHOW: 'chiTooltipShow',
+  HIDE: 'chiTooltipHide',
+  SHOWN: 'chiTooltipShown',
+  HIDDEN: 'chiTooltipHidden'
+};
+const ACTIVE_CLASS = '-active';
+
+const hasClassAssertion = (el, value) => {
+  cy.get(el).should('have.class', value);
+};
+
+describe('Pagination', () => {
+  before(() => {
+    cy.visit('tests/chi-vue/tooltip.html');
+  });
+
+  describe('Base', () => {
+    beforeEach(() => {
+      cy.get("[data-cy='tooltip-base']")
+        .children()
+        .last()
+        .as('tooltip');
+    });
+
+    it(`should have class ${TOOLTIP_CLASSES.TOOLTIP_ELEMENT}`, () => {
+      hasClassAssertion('@tooltip', `${TOOLTIP_CLASSES.TOOLTIP_ELEMENT}`);
+    });
+
+    it(`should have top position by default`, () => {
+      cy.get('@tooltip').should('have.attr', 'data-popper-placement', 'top');
+    });
+
+    it('should contain a tooltip message', () => {
+      cy.get('@tooltip').should('contain', 'Your tooltip text on a button');
+    });
+
+    it(`should have class ${ACTIVE_CLASS} when the tooltip is visible`, () => {
+      cy.get("[data-cy='tooltip-base'] button")
+        .click()
+        .then(() => {
+          hasClassAssertion('@tooltip', `${ACTIVE_CLASS}`);
+        });
+      cy.get('h3')
+        .first()
+        .click({ force: true });
+    });
+
+    it(`should trigger the tooltip events`, () => {
+      cy.window()
+        .its('baseTooltipExample')
+        .then(baseTooltipExample => {
+          const component = baseTooltipExample.$refs.baseTooltipRef;
+          const showSpy = cy.spy();
+          const shownSpy = cy.spy();
+          const hideSpy = cy.spy();
+          const hiddenSpy = cy.spy();
+
+          component.$on(`${TOOLTIP_EVENTS.SHOW}`, showSpy);
+          component.$on(`${TOOLTIP_EVENTS.SHOWN}`, shownSpy);
+          component.$on(`${TOOLTIP_EVENTS.HIDE}`, hideSpy);
+          component.$on(`${TOOLTIP_EVENTS.HIDDEN}`, hiddenSpy);
+          cy.get("[data-cy='tooltip-base'] button")
+            .trigger('mouseenter')
+            .then(() => {
+              cy.wait(1000);
+              cy.get("[data-cy='tooltip-base'] button")
```

Comment:
Can we avoid repetition of the selector? as('tooltipTrigger') or something like that.